### PR TITLE
Fix missing clinical data NA counts which have been filtered out

### PR DESCRIPTION
**Pull request by alisman**
Fixers https://github.com/cBioPortal/cbioportal/issues/11159

### CODE REVIEW

File: src/main/java/org/cbioportal/web/columnar/StudyViewColumnStoreController.java [hunk 1]
```diff
@@ -213,7 +215,26 @@ public ResponseEntity<List<ClinicalDataCountItem>> fetchClinicalDataCounts(
        // boolean singleStudyUnfiltered = studyViewFilterUtil.isSingleStudyUnfiltered(studyViewFilter);
         List<ClinicalDataCountItem> result = studyViewColumnarService.getClinicalDataCounts(studyViewFilter, 
             attributes.stream().map(ClinicalDataFilter::getAttributeId).collect(Collectors.toList()));
-            //studyIds, sampleIds, attributes.stream().map(a -> a.getAttributeId()).collect(Collectors.toList())); 
+
+        // fetch the samples by using the provided study view filter
+        List<Sample> filteredSamples = studyViewColumnarService.getFilteredSamples(studyViewFilter);
```
Comment (haynescd) [inline src/main/java/org/cbioportal/web/columnar/StudyViewColumnStoreController.java:220]: I would push this business logic down into the service (eventually business logic shouldn't even be at this layer)

and utilize the studyViewRepository.getFilteredSamplesCount().

Probably wrap this in a function.

File: src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java [hunk 2]
```diff
@@ -153,8 +154,32 @@ public Map<String, ClinicalDataType> getClinicalAttributeDatatypeMap(StudyViewFi
...
+                .collect(Collectors.groupingBy(ClinicalDataCount::getValue));
+            List<ClinicalDataCount> res = countsPerType.entrySet().stream().map((entry)->{
+                ClinicalDataCount mergedCount = new ClinicalDataCount();
+                mergedCount.setAttributeId(attr.getAttributeId());
+                mergedCount.setValue(entry.getKey());
+                mergedCount.setCount(entry.getValue().stream().mapToInt(ClinicalDataCount::getCount).sum());
+                return mergedCount;
+            }).collect(Collectors.toList());
+            attr.setCounts(res);
+        });
```
Comment (haynescd) [inline src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java:176]: Did you mean to use StudyViewColumnarServiceUtil.mergeClinicalDataCounts?

File: src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java [hunk 3]
```diff
@@ -153,8 +154,32 @@ public Map<String, ClinicalDataType> getClinicalAttributeDatatypeMap(StudyViewFi
     )
     @Override
     public List<ClinicalDataCountItem> getClinicalDataCounts(StudyViewFilter studyViewFilter, List<String> filteredAttributes) {
-       return studyViewRepository.getClinicalDataCounts(createContext(studyViewFilter), filteredAttributes);
+
+        var result = studyViewRepository.getClinicalDataCounts(createContext(studyViewFilter), filteredAttributes);
+        
+        // fetch the samples by using the provided study view filter
+        List<Sample> filteredSamples = getFilteredSamples(studyViewFilter);
```
Comment (haynescd) [inline src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java:161]: I would still suggest this studyViewRepository.getFilteredSamplesCount()

**Review by haynescd — CHANGES_REQUESTED** [review]

File: src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java (hunk 2 above)
Comment (alisman) [reply src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java:176]: good point 😆

File: src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java [hunk 4]
```diff
@@ -153,8 +154,33 @@ public Map<String, ClinicalDataType> getClinicalAttributeDatatypeMap(StudyViewFi
...
+//                .collect(Collectors.groupingBy(ClinicalDataCount::getValue));
+//            List<ClinicalDataCount> res = countsPerType.entrySet().stream().map((entry)->{
+//                ClinicalDataCount mergedCount = new ClinicalDataCount();
+//                mergedCount.setAttributeId(attr.getAttributeId());
+//                mergedCount.setValue(entry.getKey());
+//                mergedCount.setCount(entry.getValue().stream().mapToInt(ClinicalDataCount::getCount).sum());
+//                return mergedCount;
+//            }).collect(Collectors.toList());
+//            attr.setCounts(res);
+//        });
```
Comment (haynescd) [inline src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java:178]: Can you remove?

**Review by haynescd — CHANGES_REQUESTED** [review]

File: src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java (hunk 3 above)
Comment (alisman) [reply src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java]: @haynescd my understanding was just that you wanted me to call the repository directly, right here

**Pull request closed without merging**